### PR TITLE
UI improvements

### PR DESCRIPTION
**Pull request by pulsejet**
WIP
Testing notes: clear vite cache (`node_modules/.vite`) and `pnpm install` again.

### CODE REVIEW

**Comment by pulsejet** [comment]
One blocking issue here: `patch-package` doesn't work with `pnpm`.

File: src/workers/sw.ts [hunk 1]
```diff
@@ -108,9 +108,9 @@ registerRoute(/\/stored\/.*/, async (options) => {
 });
 
 registerRoute(
-  /^https:\/\/cdn.jsdelivr.net\/npm/,
+  /^https?:\/\/(cdn\.jsdelivr\.net|texlive2\.swiftlatex\.com)/,
```
Comment (zjkmxy) [inline src/workers/sw.ts:111]: I'm fine with caching them but just make sure you know that in the deployed version, LaTeX dependencies are not served by this cache.

File: src/components/common.css [hunk 2]
```diff
@@ -17,4 +17,9 @@
 .sl-panel {
     height: 100%;
     flex: 1;
+}
+
+/** PDF Slick component */
+.pdfSlick {
```
Comment (zjkmxy) [inline src/components/common.css:23]: CSS for specific page can be loaded in place:
```typescript
import cssClassName from './pdfSlick.css'
```
Then, `cssClassName` would be a string containing the correct CSS class name.

File: src/components/share-latex/share-latex/index.tsx [hunk 3]
```diff
@@ -206,21 +206,17 @@ export default function ShareLatex(props: {
     // Check if PDF is generated
     if (!res.pdf) {
       alert('Failed to compile PDF file')
-      // setCompilationLog(res.log)
+      setCompilationLog(res.log)
```
Comment (zjkmxy) [inline src/components/share-latex/share-latex/index.tsx:209]: The log has already been set. Please remove this line.

File: src/workers/sw.ts (hunk 1 above)
Comment (pulsejet) [reply src/workers/sw.ts:111]: Hmm why so?

File: src/components/common.css (hunk 2 above)
Comment (pulsejet) [reply src/components/common.css:23]: Interesting. I do want to switch to SCSS asap; writing plain CSS is nightmarish

Comment (zjkmxy) [reply src/components/common.css:23]: That's also good.

File: src/workers/sw.ts (hunk 1 above)
Comment (zjkmxy) [reply src/workers/sw.ts:111]: Here I set the path to be our own website: https://github.com/UCLA-IRL/ndn-workspace-solid/blob/38f1b0f0184871240e047f41d0f852995076a14a/src/components/share-latex/share-latex/index.tsx#L179-L182
I think we can have more control if we use OPFS. For example, when we want to support something like loading all packages in a zip file, we will be able to do so. Also, I don't think those files will be updated frequently.

**Comment by zjkmxy** [comment]
You have already pushed too much to this PR and made it very hard to review.
Please stop pushing and mark it ready to review ASAP.
I will close this PR next Friday if it is not ready by then.

**Comment by pulsejet** [comment]
@zjkmxy comments here?

**Review by zjkmxy — APPROVED** [review]
LGTM

**Pull request merged**